agent-bootstrap/src/main/java/io/opentelemetry/auto/bootstrap/instrumentation/decorator/DatabaseClientTracer.java

```diff
@@ -40,37 +47,60 @@ public DatabaseClientTracer() {
     tracer = OpenTelemetry.getTracerProvider().get(getInstrumentationName(), getVersion());
   }
 
-  // TODO make abstract when implemented in all subclasses
-  protected String getInstrumentationName() {
-    return null;
-  }
-
-  private String getVersion() {
-    return null;
-  }
-
-  public Scope withSpan(Span span) {
-    return tracer.withSpan(span);
-  }
-
   public Span startSpan(CONNECTION connection, QUERY query, String originType) {
     String normalizedQuery = normalizeQuery(query);
 
+    return startSpan(normalizedQuery, connection, originType);
+  }
+
+  public Span startSpan(String spanName, CONNECTION connection, String originType) {
     final Span span =
         tracer
-            .spanBuilder(spanName(normalizedQuery))
+            .spanBuilder(spanName(spanName))
             .setSpanKind(CLIENT)
             .setAttribute(SemanticAttributes.DB_TYPE.key(), dbType())
             .setAttribute("span.origin.type", originType)
```

Comment:
I think we can be setting null here which is surprising, but same as previous code

Reply:
Setting null will remove attribute, so no harm done. But I really dislike this attribute :)

Reply:
I will open an issue to discuss/remove `span.origin.type` 👍

Reply:
For me this is part of #602